javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named getNames that, given an array of people objects, uses map to return an array of names reversed.
+
+For example:
+[
+{
+  name: 'lloyd',
+  age: 32,
+  shoeSize: 12
+},
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => {
+  // Solution code here...
+  const reversed = arr.map((names) => {
+    let objNames = names.name;
+    return objNames.split('').reverse().join('');
+  });
+  return reversed;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function that appends ' The end.' to a string, and returns the modified string. The original source string should not be modified.
+
+------------------------------------------------------------------------------------------------ */
+
+const appendTheEnd = (str) => {
+  // Solution code here...
+  const end = ' The end.';
+  const begin = str;
+  const joinString = begin + end;
+  return joinString;
```

Comment:
I'm not sure if this is more clear to you, but it is an awful lot of extra code.
```js
return `${str} The end.`;
```

Reply:
You can also use concatanation which may be more familiar to you, although I'd definitely encourage you to use template literals (${variable} <--- this notation). Either way, I think the impulse to give everything a semantic variable isn't necessarily bad, especially as you're approaching more complex problems, however, in this case, I think it's over complicating things.

Reply:
Also, as a side note: make sure you clean up your code. On line 42, you still have a comment telling you to write you code here. You did it, you wrote code there, so get rid of the comment.